cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -344,16 +344,17 @@ std::pair<VersionedItem, FrameAndDescriptor> LocalVersionedEngine::read_datafram
         if(opt_false(read_options.incompletes_)) {
             log::version().warn("No index:  Key not found for {}, will attempt to use incomplete segments.", stream_id);
             identifier = stream_id;
+        } else {
+            missing_data::raise<ErrorCode::E_NO_SUCH_VERSION>(
+                    "read_dataframe_version: version query '{}' not found for symbol '{}'", version_query, stream_id);
```

Comment:
How can you not find a query? Does this mean no version matching the query or something like that?

Reply:
Yes that's what it means

Reply:
...can we change the wording to make that clearer?